src/Client/GuzzleFileFetcher.php

```diff
@@ -96,14 +108,23 @@ private function onFulfilled(string $fileName, int $maxBytes): \Closure
     {
         return function (ResponseInterface $response) use ($fileName, $maxBytes) {
             $body = $response->getBody();
-            $contents = $body->read($maxBytes);
+            $size = $body->getSize();
 
-            // If we reached the end of the stream, we didn't exceed the maximum
-            // number of bytes.
-            if ($body->eof() === true) {
-                return $contents;
+            if (isset($size)) {
```

Comment:
I am unclear when this would not be set?

Reply:
`StreamInterface` makes an allowance for streams whose size is not known:
```php
    /**
     * Get the size of the stream if known.
     *
     * @return int|null Returns the size in bytes if known, or null if unknown.
     */
    public function getSize();
```
So in this case, I'm just respecting the documented behavior of the API.